game/src/test/java/com/hiddenswitch/spellsource/WitchDoctorTests.java

```diff
@@ -208,4 +213,253 @@ public void testWitchingTraveler() {
+			SpellDesc spell = new SpellDesc(ChangeHeroPowerSpell.class);
+			spell.put(SpellArg.CARD, "hero_power_hone_reflexes");
+			context.getLogic().castSpell(player.getId(), spell, player.getReference(), null, false);
+			context.getLogic().endOfSequence();
+			Card heropower = player.getHero().getHeroPower();
+			assertEquals(heropower.getCardId(), "hero_power_hone_reflexes");
+			player.setMana(20);
+			assertTrue(context.getLogic().canPlayCard(player.getId(), heropower.getReference()));
+			useHeroPower(context, player);
+			assertFalse(context.getLogic().canPlayCard(player.getId(), heropower.getReference()));
+			Minion sprite = playMinionCard(context, player, "minion_shy_sprite");
+			context.endTurn();
+			player.getHero().setHp(10);
+			playCard(context, opponent, "spell_fireball", sprite);
+			assertFalse(sprite.isDestroyed());
+			assertEquals(player.getHero().getHp(), 4);
+			context.endTurn();
+			player.setMana(20);
+			assertTrue(context.getLogic().canPlayCard(player.getId(), heropower.getReference()));
+			useHeroPower(context, player);
+			assertTrue(context.getLogic().canPlayCard(player.getId(), heropower.getReference()));
+			useHeroPower(context, player);
+			assertFalse(context.getLogic().canPlayCard(player.getId(), heropower.getReference()));
+		});
+	}
+
+	// The Uncasked: "Battlecry: Draw Elementals from your deck until your hand is full.",
+	@Test
+	public void testTheUncasked() {
+		runGym(((context, player, opponent) -> {
+			assertEquals(player.getHand().size(), 0);
+			shuffleToDeck(context, player, "minion_lake_elemental");
+			shuffleToDeck(context, player, "minion_wisp");
+			shuffleToDeck(context, player, "minion_thunderhead");
+			playMinionCard(context, player, "minion_the_uncasked");
+			assertEquals(player.getHand().size(), 2);
+			assertEquals(player.getHand().get(0).getRace(), Race.ELEMENTAL);
+			assertEquals(player.getHand().get(1).getRace(), Race.ELEMENTAL);
+		}));
+	}
+
+	// Spell Hone Reflexes: "Your Hero Power becomes 'Gain +3 Attack this turn."
+	@Test
+	public void testSpellHoneReflexes() {
+		runGym(((context, player, opponent) -> {
+			assertNotEquals(player.getHero().getHeroPower().getCardId(), "hero_power_hone_reflexes");
+			playCard(context, player, "spell_hone_reflexes");
+			assertEquals(player.getHero().getHeroPower().getCardId(), "hero_power_hone_reflexes");
+		}));
+	}
+
+	// Touch of Death: "Shoot missiles equal to your hero's Attack that each deal as much damage."
+	@Test
+	public void testTouchofDeath() {
+		runGym(((context, player, opponent) -> {
+			context.endTurn();
+			Minion target1 = playMinionCard(context, opponent, "minion_neutral_test_14");
+			context.endTurn();
+			opponent.getHero().setHp(opponent.getHero().getBaseHp());
+			player.getHero().setAttack(2);
+			playCard(context, player, "spell_touch_of_death");
+			assertEquals(opponent.getHero().getHp() + target1.getHp(), opponent.getHero().getBaseHp() + target1.getBaseHp() - 4);
+		}));
```

Comment:
Make sure to test this with spellpower. E.g. if you have +1 Spellpower and 2 champion attack, you should shoot 2 missiles that each deal (2+1)=3 damage.

Reply:
What's spellpower? :o